### ISSUE TITLE
HGrid Keyboard Navigation does not work correctly when there is an expanded 3-rd level child grid with no data

### ISSUE THREAD

**Issue by PavlovVasil** [IgniteUI/igniteui-angular#5251]
## Description  
Using the Down Arrow key on the parent grid focuses and scrolls to the wrong cell in the child grid.

 * igniteui-angular version: 7.3.x, 8.0.x
 * browser: n/a

## Steps to reproduce  

1. Run the dev demos and open http://localhost:4200/hierarchicalGridUpdating
2. Expand the first parent row "ALFKI"
3. Expand the first row of the child grid - "0"
4. Click the "ALFKI" cell in the parent grid
5. Press Down Arrow key 11 times

## Result  
On the 11-th keypress the grid incorrectly scroll upwards and selects the last row of the 2-nd level child grid.

## Expected result  
Navigation should continue downwards in the parent hierarchical grid.

## Attachments  

![HGridKeyboardNav](https://user-images.githubusercontent.com/18643062/60349311-00f4d680-99ca-11e9-843b-81298f7e31a0.gif)

**Comment by skrustev** [comment]
The issue is reproduced also if just the first grid is expanded and when navigating down the grid needs to scroll to next row. If the next row index is present in the expanded grid, it will focus that next row since it search the first row with the target index, without filtering the correct grid.
